### ISSUE TITLE
Suboptimal error message

### ISSUE THREAD

**Issue by VladUreche** [miniboxing/miniboxing-plugin#76]
```
sun@sun-laptop:/mnt/data-local/Work/Workspace/dev/miniboxing-plugin/sandbox(wip)$ cat crasher2.scala 
package miniboxing.tests.compile

class Hello[@miniboxed T] {
  case class Boo(x: Int)
  def foo(b: Boo): Unit = ???
}

sun@sun-laptop:/mnt/data-local/Work/Workspace/dev/miniboxing-plugin/sandbox(wip)$ ../mb-scalac -P:minibox:spec-no-opt crasher2.scala 
crasher2.scala:4: error: The program is accessing field Boo$module of miniboxed class (or trait) Hello, a pattern which becomes invalid after the miniboxing transformation. Please allow Scala to generate getters (and possibly setters) by using val (or var) without the "private[this]" qualifier: var Boo$module: Hello.this.Boo.type".
  case class Boo(x: Int)
             ^
one error found
sun@sun-laptop:/mnt/data-local/Work/Workspace/dev/miniboxing-plugin/sandbox(wip)$ 
```

**Comment by VladUreche** [comment]
The message might actually be completely incorrect. Needs investigation.

**Comment by VladUreche** [comment]
Yes, the message is incorrect. Yet, correcting the condition for generating the message reveals a real issue:

```
$ ../mb-scalac gh-bug-76.scala -Ycheck:uncurry
[Now checking: uncurry]
gh-bug-76.scala:4: error: 
**** ERROR DURING INTERNAL CHECKING ****
Hello.this.type does not take parameters
  case class Boo(x: Int)
             ^
one error found
```

That's a bug in scalac. Yet, I did not lodge a ticket since it's fixed in 2.11:

```
$ cat x.scala
package miniboxing.tests.compile

class Hello[T] {
  case class Boo(x: Int)
  def foo(b: Boo): Unit = ???
}

$ $SCALA_PATH_211/scalac x.scala -Ycheck:uncurry
[Now checking: uncurry]
```

Ping #101!
